src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -915,6 +915,17 @@ private static bool HadLambdaConversionError(DiagnosticBag diagnostics, Analyzed
             RefKind refArg = arguments.RefKind(arg);
             RefKind refParm = parameter.RefKind;
 
+            if (arguments.IsExtensionMethodThisArgument(arg))
+            {
+                Debug.Assert(refArg == RefKind.None);
+                if (refParm == RefKind.Ref || refParm == RefKind.RefReadOnly)
```

Comment:
Will add a comment explaining why we don't want to do that for other types.

---
In reply to: [122018245](https://github.com/dotnet/roslyn/pull/19703#discussion_r122018245) [](ancestors = 122018245,122016776)